services/getstream/followTopic.js

```diff
@@ -49,24 +49,56 @@ const addTopicToChatTab = async (token, topicName, userId, isAnonymous) => {
     const text = isAnonymous
       ? 'This topic has new anonymous followers'
       : 'This topic has new followers';
-    const defaultImage = isAnonymous ? DEFAULT_TOPIC_PIC_PATH_ANON : DEFAULT_TOPIC_PIC_PATH_SIGN;
+
+    const admin = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
+    const adminChannel = admin.channel('topics', `topic_${topic}`);
+    const adminChannelData = await adminChannel.create();
+
+    const updateData = {};
+    // Revert back all existing signed channel that has anon topic to correct signed one
+    if (
+      [DEFAULT_TOPIC_PIC_PATH_ANON, DEFAULT_TOPIC_PIC_PATH_SIGN].includes(
+        adminChannelData?.channel?.image
+      )
+    ) {
+      updateData.image = null;
+    }
+
+    if (
+      [DEFAULT_TOPIC_PIC_PATH_ANON, DEFAULT_TOPIC_PIC_PATH_SIGN].includes(
+        adminChannelData?.channel?.channelImage
+      )
+    ) {
+      updateData.channelImage = null;
+    }
+
+    if (
+      [DEFAULT_TOPIC_PIC_PATH_ANON, DEFAULT_TOPIC_PIC_PATH_SIGN].includes(
+        adminChannelData?.channel?.channel_image
+      )
+    ) {
+      updateData.channel_image = null;
+    }
+    // Revert END
+
+    await adminChannel.updatePartial({
+      set: updateData
+    });
 
     const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
     await client.connectUser({id: userId}, token);
     const channel = await client.channel('topics', `topic_${topic}`, {
       name: `#${topic}`,
       members: [userId],
-      channel_type: 3,
-      channel_image: defaultImage,
-      channelImage: defaultImage,
-      image: defaultImage
+      channel_type: 3
     });
 
     console.log('prepare follow channel', topic);
     await channel.create();
     await channel.addMembers([userId]);
     await channel.sendMessage({text}, {skip_push: true});
     console.log('channel followed');
+    await client.disconnectUser();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is no error handling for the async operations in these lines. If any of these promises reject, it will result in an unhandled promise rejection which can lead to unpredictable behavior. It's recommended to wrap these operations in a try-catch block or handle the promise rejection with `.catch()`.

```diff
+    try {
         console.log('prepare follow channel', topic);
         await channel.create();
         await channel.addMembers([userId]);
         await channel.sendMessage({text}, {skip_push: true});
         console.log('channel followed');
+    } catch (error) {
+        console.error('Error while preparing the follow channel:', error);
+    }
```

Overall, the changes seem to be addressing the issue described in the PR title. However, the above improvements would enhance the maintainability and reliability of the code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->